src/libs/blueprint/conduit_blueprint_mesh_examples.cpp

```diff
@@ -2532,6 +2532,175 @@ void spiral(index_t ndoms,
+    res["fields/circle_c/topology"] = "topo";
+    res["fields/circle_c/values"] = DataType::float64(nx * ny);
+
+    // per element how many circles overlap
+    res["fields/overlap/association"] = "element";
+    res["fields/overlap/topology"] = "topo";
+    res["fields/overlap/values"] = DataType::float64(nx * ny);
+
+
+    float64_array rad_a = res["fields/radius_a/values"].value();
+    float64_array cir_a = res["fields/circle_a/values"].value();
+    
+    float64_array rad_b = res["fields/radius_b/values"].value();
+    float64_array cir_b = res["fields/circle_b/values"].value();
+
+    float64_array rad_c = res["fields/radius_c/values"].value();
+    float64_array cir_c = res["fields/circle_c/values"].value();
+
+
+    float64_array olap  = res["fields/overlap/values"].value();
+        
+    // circle a
+    // centered at:
+    //   x = 2/3 of width
+    //   y = 2/3 of width
+    
+    float64 a_center_x = 0.3333333333;
+    float64 a_center_y = 0.6666666666;
+
+    // circle b
+    // centered at:
+    //   x = 2/3 of width
+    //   y = 2/3 of width
+
+    float64 b_center_x = 0.6666666666;
+    float64 b_center_y = 0.6666666666;
+
+    // circle c
+    // centered at:
+    //   x = 1/2 of width
+    //   y = 2/3 of width
+    float64 c_center_x = 0.5;
+    float64 c_center_y = 0.3333333333;
+
+
+    float64 y = 0;
+    
+    index_t idx = 0;
+    for(index_t j = 0; j < ny; j++)
+    {
+        float64 x = 0;
+        for(index_t i = 0; i < nx; i++)
+        {
+            // dist to circle a
+            rad_a[idx] = sqrt( (x - a_center_x)*(x - a_center_x) +
+                               (y - a_center_y)*(y - a_center_y));
+            if(rad_a[idx] > radius)
+                rad_a[idx] = 0.0; // clamp outside of radius
+
+            // dist to circle b
+            rad_b[idx] = sqrt( (x - b_center_x)*(x - b_center_x) +
+                               (y - b_center_y)*(y - b_center_y));
+            if(rad_b[idx] > radius)
+                rad_b[idx] = 0.0; // clamp outside of radius
+
+            // dist to circle c
+            rad_c[idx] = sqrt( (x - c_center_x)*(x - c_center_x) +
+                               (y - c_center_y)*(y - c_center_y));
+            if(rad_c[idx] > radius)
+                rad_c[idx] = 0.0; // clamp outside of radius
+
+            // populate overlap with count
+            if(rad_a[idx] > 0)
+                olap[idx]++;
+
+            if(rad_b[idx] > 0)
+                olap[idx]++;
+
+            if(rad_c[idx] > 0)
+                olap[idx]++;
+
+            // circle vfs
+            if(rad_a[idx] > 0)
+                cir_a[idx] = 1.0/olap[idx];
+            else
+                cir_a[idx] = 0.0;
+
+            if(rad_b[idx] > 0)
+                cir_b[idx] = 1.0/olap[idx];
+            else
+                cir_b[idx] = 0.0;
+
+            if(rad_c[idx] > 0)
+                cir_c[idx] = 1.0/olap[idx];
+            else
+                cir_c[idx] = 0.0;
```

Comment:
If all of the circle centers (e.g. `a_center_*`) and their associated arrays (e.g. `cir_a`, `rad_a`) were put into parallel arrays of their own, this loop body could be transformed into a loop of its own over the individual circles, which would help eliminate some code duplication. If we choose to do this, it would probably help to name the circles by their indices in this array (e.g. `circle_0` instead of `circle_a`) in order to improve the clarity of the code.

Reply:
i think this logic is a bit verbose, but I think its easier to comeback to cold to understand the example.

Recently revisiting the nest set examples, extra looping make it harder to dust off.  So prefer the simpler but longer formation.